### DISCUSSION TITLE
issue: conditional validation with zod `.refine` only works on blur

### DISCUSSION THREAD

**Discussion by rodrigovallades** [react-hook-form/react-hook-form#12125] [Q&A]
### Version Number

7.52.1

### Codesandbox/Expo snack

https://codesandbox.io/p/sandbox/gracious-fire-8cvgq6?file=%2Fsrc%2FForm.tsx

### Steps to reproduce

1. Clicking on the `Disable check` checkbox should bypass form validation
2. But the form only becomes valid after the checkbox field is blurred (click outside, navigate with tab on the keyboard, etc).

### Expected behaviour

Form validation mode is set to `all`.

Am I missing something here? I expect the form to become valid as soon as I enable the `Disable check` checkbox.

### What browsers are you seeing the problem on?

Chrome

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
```
resolver: async (data, context, options) => {
  // you can debug your validation schema here
  console.log("formData", data)
  console.log(
    "validation result",
    await anyResolver(schema)(data, context, options)
  )
  return anyResolver(schema)(data, context, options)
}
```

you can use the code snippet above to debug the root cause, hook form just get validation result after schema resolved

![Screenshot 2024-07-23 at 7 05 32 pm](https://github.com/user-attachments/assets/8c1b2762-11a9-45a5-9c36-2b8d876b24cf)


https://react-hook-form.com/docs/useform
